lib/node_modules/@stdlib/blas/ext/base/sindex-of/lib/main.js

```diff
@@ -0,0 +1,76 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2025 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+// MAIN //
+
+/**
+* Returns the index of a specified search element in a single-precision floating-point strided array.
+*
+* @param {PositiveInteger} N - number of indexed elements
+* @param {Number} searchElement - number to search index of
+* @param {Float32Array} x - input array
+* @param {integer} strideX - index increment
+* @param {NonNegativeInteger} [offsetX] - starting index
+* @returns {integer} index
+*
+* @example
+* var Float32Array = require( '@stdlib/array/float32' );
+*
+* var x = new Float32Array( [ -2.0, 1.0, 3.0, -5.0, 4.0, 0.0, -1.0, 3.0 ] );
+*
+* var idx = sindexOf( x.length, 3.0, x, 1 );
+* // returns 2
+*/
+function sindexOf( N, searchElement, x, strideX, offsetX ) {
+	var ix;
+	var i;
+
+	if ( N <= 0 || x.length === 0 ) {
+		return -1;
+	}
+
+	if ( offsetX === void 0 ) {
+		if ( strideX < 0 ) {
+			ix = x.length - 1;
```

Comment:
We should NOT be assuming or inquiring anything about `x` length. Plus, this doesn't make sense at all. `N` and `stride` completely specify the array view defining what values are indexed.